snap/snapcraft.yaml

```diff
@@ -0,0 +1,204 @@
+name: docker
+version: 17.03.1-ce-0
+summary: Docker Linux container runtime
+description: Docker complements kernel namespacing with a high-level API which operates at the process level. It runs unix processes with strong guarantees of isolation and repeatability across servers.
+
+confinement: strict
+grade: stable
+
+plugs:
+  support:
+    interface: docker-support
+  privileged:
+    interface: docker-support
+    privileged-containers: true
+  docker-cli:
+    interface: docker
+slots:
+  docker-daemon:
+    interface: docker
+
+apps:
+  docker:
+    command: docker-wrapper docker
+    plugs:
+      - docker-cli
+      - network
+      - home
+
+  dockerd:
+    command: dockerd-wrapper
+    daemon: simple
+    plugs:
+      - network-bind
+      - firewall-control
+      - support
+      - privileged
+    slots:
+      - docker-daemon
+
+  # TODO determine if we can "provide" a "docker-compose" in PATH directly (since this'll show up as "docker.compose" in the user PATH)
+  compose:
+    command: docker-wrapper docker-compose
+    plugs:
+      - docker-cli
+      - network
+      - home
+  help:
+    command: help
+
+parts:
+  snappy-bins:
+    plugin: shell
+    shell: bash
+    shell-flags: ['-ex']
+    shell-command: |
+      install -d "$SNAPCRAFT_PART_INSTALL/bin"
+      install -t "$SNAPCRAFT_PART_INSTALL/bin" "$SNAPDIR"/bin/*
+    stage-packages:
+      - mount
+
+  go:
+    plugin: gobuild
+    source: https://storage.googleapis.com/golang/go1.7.5.src.tar.gz
+    source-type: tar
+    snap:
+      - -*
+    build-packages:
+      - golang-go
+      - rsync
+      - build-essential
+      - netbase
+      - python3-pyelftools
```

Comment:
Right, golang1.7 is not available from official ppa on Xenial(16.04). Add the golang1.7 as a build dependency only works for Zesty(17.04), which is not good.
https://github.com/docker/docker-snap/blob/master/snapcraft.yaml#L98

As you can see on another feature branch Mike imported 3rd-party ppa to install golang1.7.
https://github.com/docker/docker-snap/blob/fixup-apt-access/snapcraft.yaml#L69
However, it only works for a few architectures AFAIK. golang1.7_armhf can not be found on the imported ppa which failed to build the snap against armhf on launchpad.  

On the other hand, go plugin doesn't support golang1.7 yet as of writing. 
https://bugs.launchpad.net/snapcraft/+bug/1616985
Also, I found that golang1.8 is required for recent docker build. 
So building Go from source is the flexible way to get the snap build against various architectures for now.